blockchain/blockchain.go

```diff
@@ -2128,8 +2141,15 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 		return nil, 0, err
 	}
 	if vmConfig.EnableInternalTxTracing {
-		// TODO-InternalTxTracer
-		// Need to store traces to the database or send them to the designated channel
+		switch tracer := vmConfig.Tracer.(type) {
+		case *vm.InternalTxTracer:
+			result, err := tracer.GetResult()
+			if err != nil {
+				logger.Error("failed to get tracing result from a transaction", "txHash", tx.Hash().String(), "err", err)
+				return nil, 0, err
+			}
+			bc.internalTraces = append(bc.internalTraces, result)
+		}
```

Comment:
What will be happened with unknown type?

Reply:
Nothing. I put a warning log instead. 64ed21931e930a8d362d718f9909f34c62ccd513